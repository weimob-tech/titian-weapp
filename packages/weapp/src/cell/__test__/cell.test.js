const { getComponent } = require('../../common/test/getComponent');

describe('test component', () => {
  it('should render success', () => {
    const comp = getComponent('cell');
    expect(comp).toMatchSnapshot();
  });

  it('should render title width', () => {
    const comp = getComponent('cell', {
      titleWidth: '168rpx',
      title: '测试测试测试测试测试测试测试测试测试测试测试测试'
    });
    expect(comp).toMatchSnapshot();
  });

  it('should render success with title, label, desc, sub-desc', () => {
    const comp = getComponent('cell', {
      title: 'title',
      label: 'label',
      desc: 'desc',
      subDesc: 'sub-desc'
    });

    expect(comp).toMatchSnapshot();
  });

  it('should render success with title, icon', () => {
    const comp = getComponent('cell', {
      title: 'title',
      icon: 'checked'
    });
    expect(comp).toMatchSnapshot();
  });

  it('should render success with title, right icon', () => {
    const comp = getComponent('cell', {
      title: 'title',
      rightIcon: 'checked'
    });

    expect(comp).toMatchSnapshot();
  });

  it('should render success with required', () => {
    const comp = getComponent('cell', {
      title: 'title',
      required: true
    });
    expect(comp).toMatchSnapshot();
  });

  it('should render success without arrow', () => {
    const comp = getComponent('cell', {
      title: 'title',
      arrow: false
    });
    expect(comp).toMatchSnapshot();
  });

  it('should clickable disable', () => {
    const comp = getComponent('cell', {
      title: 'title',
      clickable: false
    });
    expect(comp).toMatchSnapshot();
  });

  it('should ext style add success', () => {
    const comp = getComponent('cell', {
      title: 'title',
      extStyle: 'background-color: red;'
    });
    expect(comp).toMatchSnapshot();
  });

  it('should tap success', (done) => {
    const comp = getComponent('cell', {
      title: 'title'
    });

    const callback = jest.fn();

    comp.addEventListener('click', callback);

    comp.querySelector('.titian-cell').dispatchEvent('tap');
    Promise.resolve().then(() => {
      expect(callback).toHaveBeenCalled();
      done();
    });
  });

  it('should tap not success when disable is true', (done) => {
    const comp = getComponent('cell', {
      title: 'title',
      disabled: true
    });

    const callback = jest.fn();

    comp.addEventListener('click', callback);

    comp.querySelector('.titian-cell').dispatchEvent('tap');
    Promise.resolve().then(() => {
      expect(callback).not.toHaveBeenCalled();
      done();
    });
  });
});
