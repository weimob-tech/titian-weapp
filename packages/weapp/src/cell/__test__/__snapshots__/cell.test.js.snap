// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should clickable disable 1`] = `
<main>
  <wx-view
    class="ext-class titian-cell-divider titian-cell titian-border-bottom"
    hoverClass="hover-class "
    hoverStayTime="60"
    style="align-items:center"
    bind:tap="onClick"
  >
    <wx-view
      class="title-wrap-class titian-cell-title-wrap"
      style="align-items:center"
    >
      <wx-view
        class="titian-cell-icon-box"
      />
      <wx-view>
        <wx-view
          style="display:flex;align-items:center"
        >
          <wx-view
            class="title-class titian-cell-title"
            style=""
          >
            
          title
        
          </wx-view>
        </wx-view>
      </wx-view>
    </wx-view>
    <wx-view
      class="titian-cell-desc-wrap"
    >
      <wx-view
        class="desc-content-class titian-cell-desc-content"
      />
    </wx-view>
  </wx-view>
</main>
`;

exports[`test component should ext style add success 1`] = `
<main>
  <wx-view
    class="ext-class titian-cell-divider titian-cell titian-border-bottom"
    hoverClass="hover-class "
    hoverStayTime="60"
    style="align-items:center;background-color: red;"
    bind:tap="onClick"
  >
    <wx-view
      class="title-wrap-class titian-cell-title-wrap"
      style="align-items:center"
    >
      <wx-view
        class="titian-cell-icon-box"
      />
      <wx-view>
        <wx-view
          style="display:flex;align-items:center"
        >
          <wx-view
            class="title-class titian-cell-title"
            style=""
          >
            
          title
        
          </wx-view>
        </wx-view>
      </wx-view>
    </wx-view>
    <wx-view
      class="titian-cell-desc-wrap"
    >
      <wx-view
        class="desc-content-class titian-cell-desc-content"
      />
    </wx-view>
  </wx-view>
</main>
`;

exports[`test component should render success 1`] = `
<main>
  <wx-view
    class="ext-class titian-cell-divider titian-cell titian-border-bottom"
    hoverClass="hover-class "
    hoverStayTime="60"
    style="align-items:center"
    bind:tap="onClick"
  >
    <wx-view
      class="title-wrap-class titian-cell-title-wrap"
      style="align-items:center"
    >
      <wx-view
        class="titian-cell-icon-box"
      />
      <wx-view>
        <wx-view
          style="display:flex;align-items:center"
        />
      </wx-view>
    </wx-view>
    <wx-view
      class="titian-cell-desc-wrap"
    >
      <wx-view
        class="desc-content-class titian-cell-desc-content"
      />
    </wx-view>
  </wx-view>
</main>
`;

exports[`test component should render success with required 1`] = `
<main>
  <wx-view
    class="ext-class titian-cell-divider titian-cell titian-border-bottom"
    hoverClass="hover-class "
    hoverStayTime="60"
    style="align-items:center"
    bind:tap="onClick"
  >
    <wx-view
      class="title-wrap-class titian-cell-title-wrap"
      style="align-items:center"
    >
      <wx-view
        class="titian-cell-icon-box"
      />
      <wx-view>
        <wx-view
          style="display:flex;align-items:center"
        >
          <wx-view
            class="title-class titian-cell-title"
            style=""
          >
            
          title
        
          </wx-view>
          <wx-view
            class="titian-cell-required-box"
          >
            <ti-icon
              extClass="titian-cell-required"
            >
              <wx-view
                class="titian-icon-base titian-icon titian-icon-required ext-class"
                style="font-size:24rpx"
                bind:tap="onClick"
              />
            </ti-icon>
          </wx-view>
        </wx-view>
      </wx-view>
    </wx-view>
    <wx-view
      class="titian-cell-desc-wrap"
    >
      <wx-view
        class="desc-content-class titian-cell-desc-content"
      />
    </wx-view>
  </wx-view>
</main>
`;

exports[`test component should render success with title, icon 1`] = `
<main>
  <wx-view
    class="ext-class titian-cell-divider titian-cell titian-border-bottom"
    hoverClass="hover-class "
    hoverStayTime="60"
    style="align-items:center"
    bind:tap="onClick"
  >
    <wx-view
      class="title-wrap-class titian-cell-title-wrap"
      style="align-items:center"
    >
      <wx-view
        class="titian-cell-icon-box"
      >
        <ti-icon
          extClass="titian-cell-icon"
        >
          <wx-view
            class="titian-icon-base titian-icon titian-icon-checked ext-class"
            style="font-size:36rpx"
            bind:tap="onClick"
          />
        </ti-icon>
      </wx-view>
      <wx-view>
        <wx-view
          style="display:flex;align-items:center"
        >
          <wx-view
            class="title-class titian-cell-title"
            style=""
          >
            
          title
        
          </wx-view>
        </wx-view>
      </wx-view>
    </wx-view>
    <wx-view
      class="titian-cell-desc-wrap"
    >
      <wx-view
        class="desc-content-class titian-cell-desc-content"
      />
    </wx-view>
  </wx-view>
</main>
`;

exports[`test component should render success with title, label, desc, sub-desc 1`] = `
<main>
  <wx-view
    class="ext-class titian-cell-divider titian-cell titian-border-bottom"
    hoverClass="hover-class "
    hoverStayTime="60"
    style="align-items:center"
    bind:tap="onClick"
  >
    <wx-view
      class="title-wrap-class titian-cell-title-wrap"
      style="align-items:center"
    >
      <wx-view
        class="titian-cell-icon-box"
      />
      <wx-view>
        <wx-view
          style="display:flex;align-items:center"
        >
          <wx-view
            class="title-class titian-cell-title"
            style=""
          >
            
          title
        
          </wx-view>
        </wx-view>
        <wx-view
          class="label-class titian-cell-label"
        >
          
        label
      
        </wx-view>
      </wx-view>
    </wx-view>
    <wx-view
      class="titian-cell-desc-wrap"
    >
      <wx-view
        class="desc-content-class titian-cell-desc-content"
      >
        <wx-view
          class="desc-class titian-cell-desc"
        >
          
        desc
      
        </wx-view>
      </wx-view>
      <wx-view
        class="sub-desc-class titian-cell-sub-desc"
        style=""
      >
        
      sub-desc
      
      </wx-view>
    </wx-view>
  </wx-view>
</main>
`;

exports[`test component should render success with title, right icon 1`] = `
<main>
  <wx-view
    class="ext-class titian-cell-divider titian-cell titian-border-bottom"
    hoverClass="hover-class "
    hoverStayTime="60"
    style="align-items:center"
    bind:tap="onClick"
  >
    <wx-view
      class="title-wrap-class titian-cell-title-wrap"
      style="align-items:center"
    >
      <wx-view
        class="titian-cell-icon-box"
      />
      <wx-view>
        <wx-view
          style="display:flex;align-items:center"
        >
          <wx-view
            class="title-class titian-cell-title"
            style=""
          >
            
          title
        
          </wx-view>
        </wx-view>
      </wx-view>
    </wx-view>
    <wx-view
      class="titian-cell-desc-wrap"
    >
      <wx-view
        class="desc-content-class titian-cell-desc-content"
      >
        <ti-icon
          extClass="titian-cell-right-icon"
        >
          <wx-view
            class="titian-icon-base titian-icon titian-icon-checked ext-class"
            style="font-size:28rpx"
            bind:tap="onClick"
          />
        </ti-icon>
      </wx-view>
    </wx-view>
  </wx-view>
</main>
`;

exports[`test component should render success without arrow 1`] = `
<main>
  <wx-view
    class="ext-class titian-cell-divider titian-cell titian-border-bottom"
    hoverClass="hover-class "
    hoverStayTime="60"
    style="align-items:center"
    bind:tap="onClick"
  >
    <wx-view
      class="title-wrap-class titian-cell-title-wrap"
      style="align-items:center"
    >
      <wx-view
        class="titian-cell-icon-box"
      />
      <wx-view>
        <wx-view
          style="display:flex;align-items:center"
        >
          <wx-view
            class="title-class titian-cell-title"
            style=""
          >
            
          title
        
          </wx-view>
        </wx-view>
      </wx-view>
    </wx-view>
    <wx-view
      class="titian-cell-desc-wrap"
    >
      <wx-view
        class="desc-content-class titian-cell-desc-content"
      />
    </wx-view>
  </wx-view>
</main>
`;

exports[`test component should render title width 1`] = `
<main>
  <wx-view
    class="ext-class titian-cell-divider titian-cell titian-border-bottom"
    hoverClass="hover-class "
    hoverStayTime="60"
    style="align-items:center"
    bind:tap="onClick"
  >
    <wx-view
      class="title-wrap-class titian-cell-title-wrap"
      style="align-items:center"
    >
      <wx-view
        class="titian-cell-icon-box"
      />
      <wx-view>
        <wx-view
          style="display:flex;align-items:center"
        >
          <wx-view
            class="title-class titian-cell-title"
            style="width: 168rpx"
          >
            
          测试测试测试测试测试测试测试测试测试测试测试测试
        
          </wx-view>
        </wx-view>
      </wx-view>
    </wx-view>
    <wx-view
      class="titian-cell-desc-wrap"
    >
      <wx-view
        class="desc-content-class titian-cell-desc-content"
      />
    </wx-view>
  </wx-view>
</main>
`;
